### DISCUSSION TITLE
Remove all default keybindings

### DISCUSSION THREAD

**Discussion by MunifTanjim** [nvim-neo-tree/neo-tree.nvim#265] [Q&A]
Is there a way to remove all default keybindings?

**Comment by cseickel** [comment]
No, but it could be added if there was a lot of interest. Right now the solution is to map the items you want mapped and let them override the defaults, and any remaining defaults that you don't want to have you can map to `"none"`.

**Reply by gegoune** [reply]
On a slightly unrelated note, why is it to be set to `'none'` rather than `nil`?

**Reply by cseickel** [reply]
The way lua tables work, assigning `nil` to an item effectively deletes it. There is no way to distinguish between a key that has `nil` explicitly assigned to it and a key that just does not exist. That means that it can't override a default value when the two are merged unless it has a value other than `nil`.

**Reply by MunifTanjim** [reply]
https://github.com/kyazdani42/nvim-tree.lua/blob/d5e4f0655b9bab94f1a4560fd66e7d9a03b7a009/doc/nvim-tree-lua.txt#L412-L415

It would be helpful if something like this was supported. It would help in situation where user only wants to have mapping for some specific features and disable all else.

I can open a PR if you're open to the idea.

**Reply by cseickel** [reply]
> I can open a PR if you're open to the idea.

Sure, I think that would be a good addition. Beware, the setup call is the ugliest and most confusing part of the code base.

**Reply by MunifTanjim** [reply]
Created the PR: https://github.com/nvim-neo-tree/neo-tree.nvim/pull/302

Let me know if you want to rename the config option name to something else.
